libs/domains/services/feature/src/lib/hooks/use-deployment-restrictions/use-deployment-restrictions.ts

```diff
@@ -0,0 +1,18 @@
+import { useQuery } from '@tanstack/react-query'
+import { type ApplicationType } from '@qovery/domains/services/data-access'
+import { queries } from '@qovery/state/util-queries'
+
+export interface UseDeploymentRestrictionsProps {
+  serviceId: string
+  serviceType: ApplicationType
+}
+
+export function useDeploymentRestrictions({ serviceId, serviceType }: UseDeploymentRestrictionsProps) {
+  return useQuery({
+    // eslint-disable-next-line @typescript-eslint/no-extra-non-null-assertion
```

Comment:
This is good rule by itself but it's a tanstack query limitation https://tkdodo.eu/blog/react-query-and-type-script#type-safety-with-the-enabled-option
I don't like the solution to loosen up the data-access part because it doesn't reflect our API.